collect_app/src/test/java/org/odk/collect/android/entities/JsonFileEntitiesRepositoryTest.kt

```diff
@@ -23,8 +26,22 @@ class JsonFileEntitiesRepositoryTest : EntitiesRepositoryTest() {
 
         val entity = Entity("stuff", "1", "A thing")
         one.save(entity)
-        assertThat(two.getDatasets(), contains("stuff"))
+        assertThat(two.getLists(), contains("stuff"))
         assertThat(two.getEntities("stuff"), contains(entity))
-        assertThat(three.getDatasets().size, equalTo(0))
+        assertThat(three.getLists().size, equalTo(0))
+    }
+
+    @Test
+    fun `deletes existing backing file if it can't be parsed by current code`() {
```

Comment:
This prevents us running into crashes when upgrading versions that use `JsonFileEntitiesRepository`. Clearing out entities on upgrade is obviously not ideal, but we don't want to go down the road of writing JSON migrations or storing the in a DB (quite yet). We'll note this as a caveat to users when the beta gets released.

Reply:
Actually, you assert that the repository no longer contains the entity but not if the file exists or not so it's not consistent with what the title says.